k8s/base/deployment.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: shiny-deploy
+  labels:
+    app: shiny
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: shiny
+  template:
+    metadata:
+      labels:
+        app: shiny
+    spec:
+      initContainers:
+        - name: init-shiny
+          image: busybox:1.28
+          command: ["sh", "-c", "mkdir -p /shiny/logs /shiny/apps"]
+          volumeMounts:
+            - name: shiny-data
+              mountPath: /shiny
+      containers:
+        - name: shiny
+          image: absternator/shiny-server:dev # todo: change to actual one!!! mrcide-
+          volumeMounts:
+            - name: shiny-data
+              mountPath: /shiny
+          # todo: create appropriate resource requests and limits
```

Comment:
What is the consequence of these limits? Are they speculative (affecting scheduling) or caps (meaning the scheduler kills processes that exceed them)?

Reply:
Have updated comments but i think usually requests are set or nothing is fine most of the time ... the most common is to set up requests so that each pod has allocated specific resources before being scheduled...